### ISSUE TITLE
Handling of malformed/empty data/series causes crash

### ISSUE THREAD

**Issue by MaxLeiter** [apexcharts/apexcharts.js#757]
I'm unable to accurately reproduce in codepen because we have a pretty large codebase regarding our charts.

This is using react-apexcharts but I believe the issue lies with this library. Sometimes, we pass null and/or undefined for data, and receive this stacktrace:

```
  28018 | if (options$$1.series) {
  28019 |   this.resetSeries(false);
  28020 | 
> 28021 |   if (options$$1.series[0].data) {
        | ^  28022 |     options$$1.series = options$$1.series.map(function (s, i) {
  28023 |       return _objectSpread({}, w.config.series[i], {
  28024 |         name: s.name ? s.name : w.config.series[i] && w.config.series[i].name,
```
reporting that `TypeError: Cannot read property 'data' of undefined` from line `options$$1.series[0].data`.

Graceful handling of invalid/empty series would be great.

**Comment by junedchhipa** [comment]
ApexCharts does handle empty series.
Unless you provide the `series`, it would be difficult to pinpoint what data causes issues.

**Comment by MaxLeiter** [comment]
Shouldn't an empty series default to `options.noData`?

**Comment by junedchhipa** [comment]
Yes, that's the default behavior.

**Comment by junedchhipa** [comment]
A minimal, reproducible example will be helpful to know where the issue lies.

**Comment by MaxLeiter** [comment]
I'm trying to create a reproducible example but _something_ is acting differently (I know how unhelpful that is).

If I print `options$$1.series` (by modifying `apexcharts.common.js`) it's an empty array before the crash (`[]`).

However, the value we supply is `{name: "", data: []}`. 
<img width="1429" alt="Screen Shot 2019-07-11 at 1 31 46 PM" src="https://user-images.githubusercontent.com/8675906/61083309-43fa7500-a3e0-11e9-9cd7-f29b1fdd9d7f.png">

**Comment by junedchhipa** [comment]
What versions of apexcharts and react-apexcharts are you using?
If not the latest, I recommend upgrading to the latest and check.

**Comment by MaxLeiter** [comment]
The latest: apexcharts 3.8.2 and react-apexcharts 1.3.3. React version 16.8.

The problem seems to be originating here (via the stacktrace): https://github.com/apexcharts/react-apexcharts/blob/master/src/react-apexcharts.jsx#L100-L120

**Comment by MaxLeiter** [comment]
Hm, from the lines above:
shouldnt 
``` 
else {
        // both might be changed
        this.chart.updateOptions(this.getConfig())
      }
```
be
```
 else {
        // both might be changed
        this.chart.updateSeries(series)
        this.chart.updateOptions(this.getConfig())
      }
```

**Comment by MaxLeiter** [comment]
Swapping the above locally didn't fix the problem for me but that still seems like a bug

**Comment by MaxLeiter** [comment]
Unsure what the problem was, but I think it was on our side because switching to redux fixed it

**Comment by groubis** [comment]
I confirm that problem has been resolved when updating from 3.8.0 to 3.8.4. Thanks!
